presto-spi/src/main/java/io/prestosql/spi/security/Identity.java

```diff
@@ -109,4 +112,50 @@ public String toString()
         sb.append('}');
         return sb.toString();
     }
+
+    public static Builder forUser(String user)
+    {
+        return new Builder(user);
+    }
+
+    public static class Builder
+    {
+        private final String user;
+        private Optional<Principal> principal = Optional.empty();
+        private Map<String, SelectedRole> roles = emptyMap();
+        private Map<String, String> extraCredentials = emptyMap();
+
+        public Builder(String user)
+        {
+            this.user = requireNonNull(user, "user is null");
+        }
+
+        public Builder withPrincipal(Principal principal)
+        {
+            return withPrincipal(Optional.of(principal));
+        }
+
+        public Builder withPrincipal(Optional<Principal> principal)
+        {
+            this.principal = requireNonNull(principal, "principal is null");
+            return this;
+        }
+
+        public Builder withRoles(Map<String, SelectedRole> roles)
```

Comment:
It might be helpful to have an `addRole(String catalog, SelectedRole role)` that we can call multiple times.